### PR TITLE
Add package.yaml template

### PR DESCRIPTION
**Pull request by gromakovsky**
Problem: in Haskell packages we often use `package.yaml` file.
Some parts of it are often shared by our repos.
Hence we want to extract them into a template, so that when we create
a new package we can use all our standard things.

Solution: add a template package.yaml. See comments for details.

### CODE REVIEW

File: templates/haskell/package.yaml [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+library:
+  source-dirs: src
+
+executables:
+  myproject:
+    main:                Main.hs
+    source-dirs:         app
+
+tests:
+  baseDAO-test:
```
Comment (rvem) [inline templates/haskell/package.yaml:106]: :007:

Comment (gromakovsky) [reply templates/haskell/package.yaml:106]: Oh lul.

File: templates/haskell/package.yaml [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+# If you have multiple packages, you may wish to extract common
+# definitions into a separate file.
+# Why we need this template:
+# • harmless default extensions
+# • a good set of warnings
+# • metadata fields that are better be set
+# • some tricks (e. g. RTS options)
+name:                myproject
+# version is 0 by default to indicate that nothing is ready
+version:             0
```
Comment (pgujjula) [inline templates/haskell/package.yaml:12]: Why not 0.1.0.0 like `stack new` and `cabal init`?

File: templates/haskell/package.yaml [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+# adjust/add more authors and maintainers if necessary
+author:              Serokell
+maintainer:          Serokell <hi@serokell.io>
+# other metadata fields we usually set:
+# • copyright
```
Comment (pgujjula) [inline templates/haskell/package.yaml:17]: why not include some default fields?
```
copyright: "{YEAR} Serokell"
license: MPL-2.0
homepage: "github.com/serokell/{PROJECT}"
bug-reports: "{AUTHOR NAME}@serokell.io"
```
Then a newcomer doesn't have to guess

File: templates/haskell/package.yaml [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+# • license, license-file
+# • git
+# • homepage
+# • bug-reports
+
```
Comment (pgujjula) [inline templates/haskell/package.yaml:22]: We can also include synopsis, category, and description fields here, I think those are pretty standard. If we include them with dummy values, then newcomers will know that they are expected to include those fields.

File: templates/haskell/package.yaml [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+# Why we need this template:
+# • harmless default extensions
+# • a good set of warnings
+# • metadata fields that are better be set
+# • some tricks (e. g. RTS options)
+name:                myproject
+# version is 0 by default to indicate that nothing is ready
+version:             0
+# adjust/add more authors and maintainers if necessary
+author:              Serokell
```
Comment (pgujjula) [inline templates/haskell/package.yaml:14]: As I understand, the author/maintainer should be people within Serokell, and the copyright holder should be Serokell itself. If that's the case, then we can put a value like `"{AUTHOR NAME}"` so people know to substitute their own name.

**Review by pgujjula — CHANGES_REQUESTED** [review]
Hi this is my first time reviewing within Serokell, so if I'm doing something incorrectly, please give me a meta-review and let me know :)

File: templates/haskell/package.yaml [hunk 6]
```diff
@@ -0,0 +1,132 @@
...
+  -Weverything
+  -Wno-missing-exported-signatures
+  -Wno-missing-import-lists
+  -Wno-missed-specialisations
+  -Wno-all-missed-specialisations
+  -Wno-unsafe
+  -Wno-safe
+  -Wno-missing-local-signatures
+  -Wno-monomorphism-restriction
+  -Wno-implicit-prelude
```
[the remark this answers is not in the document]
Comment (gromakovsky) [reply templates/haskell/package.yaml:82]: Done btw.

File: templates/haskell/package.yaml (hunk 2 above)
Comment (gromakovsky) [reply templates/haskell/package.yaml:12]: Hm, I think I never used `stack new` or `cabal init`. But I think that `0.1.0.0` should denote first released version, i. e. MVP or something like that. And `0` means "it's not ready and doesn't work yet". `0` is not supposed to be uploaded to Hackage. Usually a version is assigned when a release is made, so if you open a commit where version is `X`, it means that code there is exactly `X` or newer. So I prefer setting version 0.1.0 when first release is made.

File: templates/haskell/package.yaml (hunk 3 above)
Comment (gromakovsky) [reply templates/haskell/package.yaml:17]: I am just afraid that people may blindly copy-paste these default values into their `package.yaml` files even when they are invalid. For example, in my team the copyright owner is usually not Serokell, but our customer, also the repos are often not in `github.com/serokell`. However, I agree that having examples is a good idea. I will think what to do here and will expand it.

File: templates/haskell/package.yaml (hunk 5 above)
Comment (gromakovsky) [reply templates/haskell/package.yaml:14]: Usually our packages are developed by a team that sometimes changes, so keeping track of all authors is hard. We have some packages developed by a really small number of people where it's easy to name just a few (1-3) authors, but it's a rare case. So I think `Serokell` is a reasonable default. I will expand my comments.

**Comment by gromakovsky** [comment]
@pgujjula I have made some changes, please take a look :slightly_smiling_face:

**Review by pgujjula — APPROVED** [review]

**Comment by pgujjula** [comment]
Looks good to me!

**Comment by gromakovsky** [comment]
@rvem just want to remind you that this thing exists.

**Review by rvem — APPROVED** [review]

**Pull request merged**